File: scalafmt-interfaces/jvm/src/main/java/org/scalafmt/interfaces/RepositoryCredential.java
package org.scalafmt.interfaces;

public final class RepositoryCredential {
    public final String host;
    public final String username;
    public final String password;

    public RepositoryCredential(String host, String username, String password) {
        this.host = host;
        this.username = username;
        this.password = password;
    }

    public interface ScalafmtExtension {
        Scalafmt withRepositoryCredentials(RepositoryCredential... credentials);
    }

}

File: scalafmt-interfaces/js/src/main/scala/org/scalafmt/interfaces/RepositoryCredential.scala
package org.scalafmt.interfaces

final class RepositoryCredential(
    val host: String,
    val username: String,
    val password: String,
)

object RepositoryCredential {
  trait ScalafmtExtension {
    def withRepositoryCredentials(credentials: RepositoryCredential*): Scalafmt
  }
}
